ctypesgen/printer_python/printer.py

```diff
@@ -272,32 +283,29 @@ def print_variable(self, variable):
     
     def print_macro(self, macro):
         # important: must check precisely against None because params may be an empty list for a func macro
+        self._srcinfo(macro.src)
         if macro.params is None:
             self._print_simple_macro(macro)
         else:
             self._print_func_macro(macro)
     
-    
     def _print_simple_macro(self, macro):
-        entry = "{MN} = {ME}".format(MN=macro.name, ME=macro.expr.py_string(True))
+        entry = f"{macro.name} = {macro.expr.py_string(True)}"
         if self.options.guard_macros:
             entry = self._try_except_wrap(entry)
         self.file.write(entry)
     
-    
     def _print_func_macro(self, macro):
         self.file.write(
-            "def {MN}({MP}):\n"
-            "    return {ME}".format(
-                MN=macro.name, MP=", ".join(macro.params), ME=macro.expr.py_string(True)
-            )
+            f"def {macro.name}({', '.join(macro.params)}):"
+            f"\n    return {macro.expr.py_string(True)}"
         )
     
-    
     def print_undef(self, undef):
+        self._srcinfo(undef.src)
         name = undef.macro.py_string(False)
-        self.file.write(f"# undef {name}\n")
-        entry = f"del {name}"
+        self.file.write(f"# undef {name}")
+        entry = f"\ndel {name}"
```

Comment:
This was a blunder, causes invalid output when the undef is try-except wrapped.
Here we have to forward declare the `\n` with the comment, which is fine because there is a definite follower.

Fixed in https://github.com/pypdfium2-team/ctypesgen/commit/d2a524a850f173b35e6191a1e2b6b82bb6eba46d